json_serializable/test/integration/json_keys_example.g.dart

```diff
@@ -0,0 +1,35 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
+
+// ignore_for_file: lines_longer_than_80_chars, text_direction_code_point_in_literal, inference_failure_on_function_invocation, inference_failure_on_collection_literal
+
+part of 'json_keys_example.dart';
+
+// **************************************************************************
+// JsonSerializableGenerator
+// **************************************************************************
+
+Model _$ModelFromJson(Map<String, dynamic> json) => Model(
+      firstName: json['first-name'] as String,
+      lastName: json['LAST_NAME'] as String,
+    );
+
+class _$ModelJsonKeys {
```

Comment:
Make this `abstract final` - https://dart.dev/language/modifier-reference

And make all of the fields `static const` – just much easier, cleaner, etc.

Reply:
So, you suggest the final result will be :

`abstract final class ModelJsonKeys {
  static const String firstName = 'first-name';
  static const String lastName = 'LAST_NAME';
}`

and the use :

`test('ModelJsonKeys', () {
    expect(ModelJsonKeys.firstName, 'first-name');
    expect(ModelJsonKeys.lastName, 'LAST_NAME');
  });`
  
Since you suggested the json keys to be static, I don't see any other implementation. Is this what you were thinking?

Reply:
that's fine!